### DISCUSSION TITLE
specfem3d_globe with external (regional or global) tomographic models

### DISCUSSION THREAD

**Discussion by carltape** [SPECFEM/specfem3d#1554] [SPECFEM3D_GLOBE General]
One topic we discussed at the Toronto specfem developers workshop was external tomographic models. Currently one can:
1) load an external xyz ascii tomographic model into specfem3d
2) choose from a set of pre-existing 3D tomographic models in specfem3d_globe
3) use an external global model, defined as percent perturbation from PREM, into specfem3d_globe
From our experience, option 3) does not offer a general approach. It appears that what is needed is the ability for a user to specify a global or regional model, possibly chosen from an IRIS EMC model, and use specfem3d_globe (either the full globe or a regional chunk).

We (at UAF) are going to look into interfacing regional-scale tomographic models with specfem3d_globe. The existing utility [specfem_tomo_helper](https://github.com/thurinj/specfem_tomo_helper) points at IRIS EMC and provides an xyz ascii file for specfem3d. One option is to do something similar for the global code. A long-term option would probably be to read the EMC netcdf files directly within specfem3d, then interpolate to the GLL points defining the mesh.

At this stage, we don't want to reinvent the wheel. People have used various tomographic models in specfem3d_globe. Have people achieved this using their own utilities -- or something that is within the package? For example, when GLAD-25 is run, is the starting point the EMC version or a pre-saved set of binary files that would exactly match the GLL points of a specific global mesh? Has anyone done something like generating an xyz ascii file for the global code, analogous to what has been done for the cartesian code? Or (perhaps?) written a function to read in an IRIS EMC netcdf file?

The primary goal here is to perform regional-scale simulations in a mesh that honors the topography and ellipticity of the Earth, while having appropriate doubling layers and lower-depth cutoff -- all of which is already possible in the global code. The task now is to implement some flexible options for external tomographic models. Suggestions or experiences welcomed in this discussion!

**Comment by bvanderbeek** [comment]
@carltape We modified a prior version of the global code (before the option to set a depth limit for the mesh was implemented) to read in a regional-scale anisotropic model defined by 21-component elastic tensors + density (saved as an ascii file). The only tricky part was making sure that the specfem3d_globe and regional model coordinate systems were consistent. However, we never made a pull request for this modification. I wanted to implement this in the latest specfem3d_globe version in the next month or so but maybe someone will beat me to it or has already done it.

**Comment by ljhwang** [comment]
The ASPECT team is also working on reading in netcdf files in C++ that will ultimately allow to read IRIS models. See: https://github.com/geodynamics/aspect/pull/4767

**Comment by lsawade** [comment]
- Related SPECFEM/specfem3d_globe#810 
- Related SPECFEM/specfem3d_globe#811
